docs/example-manifests/SLEMicro/dhcp/rke2-agent.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: sample-cluster
+  name: sample-cluster
+  namespace: default
+spec:
+  clusterName: sample-cluster
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: sample-cluster
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: sample-cluster
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha1
+          kind: RKE2ConfigTemplate
+          name: sample-cluster-workers
+      clusterName: sample-cluster
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: Metal3MachineTemplate
+        name: sample-cluster-workers
+      nodeDrainTimeout: 0s
+      version: v1.25.3+rke2r1
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha1
+kind: RKE2ConfigTemplate
+metadata:
+  name: sample-cluster-workers
+  namespace: default
+spec:
+  template:
+    spec:
+      agentConfig:
+        version: v1.25.3+rke2r1
+        kubelet:
+          extraArgs:
+            - provider-id=metal3://{{ ds.meta_data.uuid }}
+        nodeName: '{{ ds.meta_data.local_hostname }}'
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3MachineTemplate
+metadata:
+  name: sample-cluster-workers
+  namespace: default
+spec:
+  template:
+    spec:
+      dataTemplate:
+        name: sample-cluster-workers-template
+      hostSelector:
+        matchLabels:
+          cluster-role: worker
+      image:
+        checksum: http://imagecache.local:8080/SLE-Micro-eib-output.raw.md5sum
+        checksumType: md5sum
```

Comment:
I would suggest we use sha256 for consistency with the [existing manifests](https://github.com/suse-edge/metal3-demo/blob/main/docs/example-manifests/dhcp/rke2-agent.yaml#L62), and also because md5 is [planned for deprecation in future](https://github.com/metal3-io/metal3-docs/blob/main/design/baremetal-operator/bmh-v1beta1.md)

Reply:
done! thanks